### PR TITLE
73178 refactor tag validators for start and stop

### PR DESCRIPTION
**Pull request by janid1967**
When using StartPreservation command when onboarding Demissie for making StopPreservation, I found that validation logic for checking IsReadyToBeXXXX could be made better to reuse same logic in Tag domain class. Before this PR "same" logic was duplicated both in Tag.cs and in validators

### CODE REVIEW

File: src/Equinor.Procosys.Preservation.Command/Validators/TagValidators/TagValidator.cs [hunk 1]
```diff
@@ -42,6 +42,7 @@ public async Task<bool> HasANonVoidedRequirementAsync(int tagId, CancellationTok
             return tag != null && tag.Requirements.Any(r => !r.IsVoided);
         }
 
+        // todo remove after codereview .. diff get crazy when removing it
```
Comment (llykkelarsen) [inline src/Equinor.Procosys.Preservation.Command/Validators/TagValidators/TagValidator.cs:45]: don't forget to remove

File: src/Equinor.Procosys.Preservation.Command/Validators/TagValidators/TagValidator.cs [hunk 2]
```diff
@@ -83,6 +84,7 @@ public async Task<bool> HasRequirementWithActivePeriodAsync(int tagId, int requi
             return requirement != null && requirement.HasActivePeriod;
         }
 
+        // todo remove after codereview .. diff get crazy when removing it
```
Comment (llykkelarsen) [inline src/Equinor.Procosys.Preservation.Command/Validators/TagValidators/TagValidator.cs:87]: and here

File: src/Equinor.Procosys.Preservation.Command/Validators/TagValidators/TagValidator.cs [hunk 3]
```diff
@@ -113,6 +115,7 @@ public async Task<bool> RequirementIsReadyToBePreservedAsync(int tagId, int requ
             return requirement != null && requirement.ReadyToBePreserved;
         }
 
+        // todo remove after codereview .. diff get crazy when removing it
```
Comment (llykkelarsen) [inline src/Equinor.Procosys.Preservation.Command/Validators/TagValidators/TagValidator.cs:118]: and here

File: src/Equinor.Procosys.Preservation.Command/Validators/TagValidators/TagValidator.cs (hunk 1 above)
Comment (janid1967) [reply src/Equinor.Procosys.Preservation.Command/Validators/TagValidators/TagValidator.cs]: done

**Pull request merged**